src/main/drivers/timer.c

```diff
@@ -849,14 +849,19 @@ uint16_t timerDmaSource(uint8_t channel)
 
 uint16_t timerGetPrescalerByDesiredMhz(TIM_TypeDef *tim, uint16_t mhz)
 {
-    // protection here for desired MHZ > timerClock???
-    if ((uint32_t)(mhz * 1000000) > timerClock(tim)) {
-        return 0;
-    }
-    return (uint16_t)(round((timerClock(tim) / (mhz * 1000000)) - 1));
+    return timerGetPrescalerByDesiredHertz(tim, mhz * 1e6);
 }
 
 uint16_t timerGetPeriodByPrescaler(TIM_TypeDef *tim, uint16_t prescaler, uint32_t hertz)
 {
     return ((uint16_t)((timerClock(tim) / (prescaler + 1)) / hertz));
 }
+
+uint16_t timerGetPrescalerByDesiredHertz(TIM_TypeDef *tim, uint32_t hertz)
+{
+    // protection here for desired hertz > SystemCoreClock???
+    if (hertz > (timerClock(tim))) {
+        return 0;
+    }
+    return (uint16_t)(round((timerClock(tim) / hertz) - 1));
```

Comment:
`timerClock(tim) / hertz` is integer division, truncating the result. round won't do anything (only convert to double and back)
`(timerClock(tim) + hertz / 2 ) / hertz)` will round positive integers correctly